lib/ansible/modules/network/vyos/vyos_firewall_interfaces.py

```diff
@@ -0,0 +1,1077 @@
+# set interfaces ethernet eth3 firewall in name 'INBOUND'
+# set interfaces ethernet eth3 firewall 'local'
+# set interfaces ethernet eth3 firewall 'out'
+#
+- name: Overrides all device configuration with provided configuration
+  vyos_firewall_interfaces:
+    config:
+      - name: 'eth3'
+        access_rules:
+          - afi: 'ipv4'
+            rules:
+              - name: 'INBOUND'
+                direction: 'out'
+    state: overridden
+#
+#
+# -------------------------
+# Module Execution Result
+# -------------------------
+#
+# "before":[
+#        {
+#            "name": "eth0"
+#        },
+#        {
+#            "access_rules": [
+#                {
+#                    "afi": "ipv4",
+#                    "rules": [
+#                        {
+#                            "direction": "out",
+#                            "name": "OUTBOUND"
+#                        }
+#                    ]
+#                },
+#                {
+#                    "afi": "ipv6",
+#                    "rules": [
+#                        {
+#                            "direction": "local",
+#                            "name": "V6-LOCAL"
+#                        }
+#                    ]
+#                }
+#            ],
+#            "name": "eth1"
+#        },
+#        {
+#            "name": "eth2"
+#        },
+#        {
+#            "access_rules": [
+#                {
+#                    "afi": "ipv4",
+#                    "rules": [
+#                        {
+#                            "direction": "in",
+#                            "name": "INBOUND"
+#                        }
+#                    ]
+#                }
+#            ],
+#            "name": "eth3"
+#        }
+#    ]
+#
+#    "commands": [
+#        "delete interfaces ethernet eth1 firewall",
+#        "delete interfaces ethernet eth3 firewall in name",
+#        "set interfaces ethernet eth3 firewall out name 'INBOUND'"
+#
+#
+#    "after": [
+#        {
+#            "name": "eth0"
+#        },
+#        {
+#            "name": "eth1"
+#        },
+#        {
+#            "name": "eth2"
+#        },
+#        {
+#            "access_rules": [
+#                {
+#                    "afi": "ipv4",
+#                    "rules": [
+#                        {
+#                            "direction": "out",
+#                            "name": "INBOUND"
+#                        }
+#                    ]
+#                }
+#            ],
+#            "name": "eth3"
+#        }
+#    ]
+#
+#
+# After state
+# ------------
+#
+# vyos@vyos:~$ show configuration commands| grep firewall
+# set firewall ipv6-name 'V6-LOCAL'
+# set firewall name 'INBOUND'
+# set firewall name 'LOCAL'
+# set firewall name 'OUTBOUND'
+# set interfaces ethernet eth3 firewall 'in'
+# set interfaces ethernet eth3 firewall 'local'
+# set interfaces ethernet eth3 firewall out name 'INBOUND'
+
+
+# Using deleted per interface name
```

Comment:
Is it possible to remove a single firewall rule from an interface?

Reply:
yeah, there is a test case deleted_single added for this.